sdk/src/main/java/io/opentelemetry/sdk/trace/RandomIdsGenerator.java

```diff
@@ -22,14 +22,11 @@
 
 final class RandomIdsGenerator implements IdsGenerator {
   private static final long INVALID_ID = 0;
+  private static final long MIN_VALID_ID = 1;
 
   @Override
   public SpanId generateSpanId() {
-    long id;
-    ThreadLocalRandom random = ThreadLocalRandom.current();
-    do {
-      id = random.nextLong();
-    } while (id == INVALID_ID);
+    long id = ThreadLocalRandom.current().nextLong(0, Long.MAX_VALUE) + MIN_VALID_ID;
```

Comment:
I am confused, this will always have the most significant bit 0, is that what we want?

Reply:
I don't see what you are saying.
I want to generate a number between 1 and Long.MAX_VALUE
as the upper limit is exclusive, and I can't use MAX_VALUE+1 as it will cause overflow, I generate a number between 0 to MAX_VALUE-1 AND add 1 to it so It's the same as generating a number between 1 to MAX_VALUE.

Reply:
Long is signed in Java, so you will always generate the signed bit 0.

Reply:
@bogdandrutu is correct (good eye). This would effectively reduce our id space by 1/2, since it no longer will generate negative ids.

Reply:
I see what you re talking about only after I read the trace id specification and notified that it is being represented as hexadecimal.

Reply:
Seems like there was no value in this PR, closing it